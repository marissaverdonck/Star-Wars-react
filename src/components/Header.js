import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { device } from './Device';

const HeaderStyle = styled.header`
	width: 100vw;
	margin: ${({ theme }) => theme.indentS} auto;
	padding: ${({ theme }) => theme.indentS};
	background-color: ${({ theme }) => theme.primaryColor};
	a {
		text-decoration: none;
		color: ${({ theme }) => theme.secondaryColorDark};
	}
`;

const HeaderTitle = styled.h1`
	font-family: ${({ theme }) => theme.fontTitles};
	font-size: ${(props) =>
		props.size === 'big'
			? ({ theme }) => theme.fontSizeM
			: ({ theme }) => theme.fontSizeS};
	margin: 0;
	padding: 0;
	max-width: ${({ theme }) => theme.sectionWidth};
	margin: 0 auto;

	@media ${device.mobileL} {
		font-size: ${(props) =>
			props.size === 'big'
				? ({ theme }) => theme.fontSizeXL
				: ({ theme }) => theme.fontSizeS};
	}

	@media ${device.laptop} {
		font-size: ${(props) =>
			props.size === 'big'
				? ({ theme }) => theme.fontSizeXXL
				: ({ theme }) => theme.fontSizeS};
	}
`;

const Header = ({ size, title }) => {
	return (
		<HeaderStyle>
			<Link to='/'>
				<HeaderTitle size={size}>{title} </HeaderTitle>
			</Link>
		</HeaderStyle>
	);
};

Header.propTypes = {
	size: PropTypes.string,
	title: PropTypes.string,
};
Header.defaultProps = {
	title: 'Star Wars Universe.',
	size: 'small',
};

export default Header;
