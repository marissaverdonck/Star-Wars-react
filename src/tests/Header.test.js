import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/Header';

it('Renders correctly when there is no title and/or size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a title', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header title="Marissa" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header size="big" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly when there is a unknown size', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header size="5" />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
