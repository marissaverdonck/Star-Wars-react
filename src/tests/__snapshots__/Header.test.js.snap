// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly when there is a size 1`] = `
<header
  className="sc-bdnylx boUdXo"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="sc-gtssRu ktzkmZ"
      size="big"
    >
      Star Wars Universe.
       
    </h1>
  </a>
</header>
`;

exports[`Renders correctly when there is a title 1`] = `
<header
  className="sc-bdnylx boUdXo"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="sc-gtssRu ktzkmZ"
      size="small"
    >
      Marissa
       
    </h1>
  </a>
</header>
`;

exports[`Renders correctly when there is a unknown size 1`] = `
<header
  className="sc-bdnylx boUdXo"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="sc-gtssRu ktzkmZ"
      size="5"
    >
      Star Wars Universe.
       
    </h1>
  </a>
</header>
`;

exports[`Renders correctly when there is no title and/or size 1`] = `
<header
  className="sc-bdnylx boUdXo"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1
      className="sc-gtssRu ktzkmZ"
      size="small"
    >
      Star Wars Universe.
       
    </h1>
  </a>
</header>
`;
